### PR TITLE
Removing internal config setup from scalding platform

### PR DESCRIPTION
**Pull request by ianoc**
1) Both Timestamp and BatchID become any val classes to erase them at runtime
 ---> Knock on effect is to have them not extend ordered. Provide an implicit enrichment to add those methods back using the ordering on the companion object
2) Remove the internal config handling from the scalding platform, delegating to the Scalding Config class instead.

### CODE REVIEW

File: summingbird-builder/src/main/scala/com/twitter/summingbird/scalding/ScaldingEnv.scala [hunk 1]
```diff
@@ -116,7 +116,7 @@ case class ScaldingEnv(override val jobName: String, inargs: Array[String])
     val scald = Scalding(name, opts)
       .withRegistrars(ajob.registrars ++ builder.registrar.getRegistrars.asScala)
       .withConfigUpdater { c =>
-        c.updated(ajob.transformConfig(c.toMap))
+        com.twitter.scalding.Config.tryFrom(ajob.transformConfig(c.toMap).toMap).get
```
Comment (johnynek) [inline summingbird-builder/src/main/scala/com/twitter/summingbird/scalding/ScaldingEnv.scala]: maybe import Config here? Looks a little strange.

File: summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/ScaldingPlatform.scala [hunk 2]
```diff
@@ -652,67 +652,45 @@ class Scalding(
...
-      System.currentTimeMillis.toString)
-    conf.set("summingbird.submitted.timestamp",
-      System.currentTimeMillis.toString)
-
-    def ifUnset(k: String, v: String) { if (null == conf.get(k)) { conf.set(k, v) } }
-    // Set the mapside cache size, this is important to not be too small
-    ifUnset("cascading.aggregateby.threshold", "100000")
+    val postConfig = config.+("summingbird.options" -> options.toString)
+      .+("summingbird.jobname" -> jobName)
+      // legacy name to match scalding
```
Comment (johnynek) [inline summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/ScaldingPlatform.scala]: I don't think this is needed anymore (I think scalding already puts in the scalding.flow.submitted.timestamp).

Comment (ianoc) [inline summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/ScaldingPlatform.scala]: Should we keep the SB one? or just drop both?

File: summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/ScaldingPlatform.scala [hunk 3]
```diff
@@ -652,67 +652,43 @@ class Scalding(
...
-    Scalding.logger.debug("Removes: {}", transformedConfig.removes)
-    Scalding.logger.debug("Updates: {}", transformedConfig.updates)
+    val initKryo = conf.getKryo match {
+      case None =>
+        new serialization.KryoHadoop(ScalaMapConfig(conf.toMap))
+      case Some(kryo) => kryo
+    }
+    conf
+      .setSerialization(
+        Left((classOf[serialization.KryoHadoop], initKryo.withRegistrar(kryoReg))), Nil)
```
Comment (johnynek) [inline summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/ScaldingPlatform.scala]: can we use the `.serSerialization(Right(...))` case if passedRegistrars.isEmpty?

File: summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/store/DirectoryBatchedStore.scala [hunk 4]
```diff
@@ -81,7 +81,7 @@ class DirectoryBatchedStore[K <: Writable, V <: Writable](val rootPath: String)(
 
         val lastBatchStatus =
           hdfsPaths.map(getFileStatus(_, conf))
-            .filter { input => input._1 && (BatchID(input._2) < exclusiveUB) }
+            .filter { input => input._1 && (BatchID(input._2.milliSinceEpoch) < exclusiveUB) }
```
Comment (johnynek) [inline summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/store/DirectoryBatchedStore.scala]: this does not seem correct to me. You shouldn't just pass a millis to a `BatchID`. That's what Batchers do. Why is this change here?

Comment (ianoc) [inline summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/store/DirectoryBatchedStore.scala]: Thats what I thought too, but actually this is what we were doing here before too. 

BatchID's constructor that I removed as part of this (had to for AnyVal erasure reasons):
 def apply(ts: Timestamp) = new BatchID(ts.milliSinceEpoch)

So this prior code here was actually doing semantically the same as this new code

Comment (johnynek) [inline summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/store/DirectoryBatchedStore.scala]: Looks like @reconditesea wrote this code. It seems like it should be `batcher.batchOf(input._2)` and while we are at it, I'd rewrite this as:

```scala
.filter { case (isGood, timestamp, _) => isGood && (batcher.batchOf(timestamp) < exclusiveUB) }
```
I guess there are no tests on this code. :/

Comment (johnynek) [inline summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/store/DirectoryBatchedStore.scala]: by the way, that `apply(ts: Timestamp)` just seems like a broken method. How did that stick around so long?

Comment (ianoc) [inline summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/store/DirectoryBatchedStore.scala]: Thats a good question, i guess we don't look at this stuff that often

**Comment by reconditesea** [comment]
uh, this hurts :( 
But yeah, I think we should just remove this file. Even I cannot quite remember why I/we want to add this DirectoryBatched store in the first place.

**Comment by johnynek** [comment]
Looks good to me. I'm slightly worried about the source incompatibility here, but the point is better control over how the job is run and that is really important and not a small issue.

**Comment by ianoc** [comment]
Awesome, many thanks for the review help!
